omegatrade-app/backend/app/controller/user.controller.js

```diff
@@ -73,37 +45,27 @@ exports.register = async function (req, res) {
  * 
  * @param {*} req contains request headers and payload businessEmail,password
  */
-exports.login = function (req, res) {
-    const body = req.body;
-    User.findUser(body.businessEmail, function (err, user) {
+exports.login = async function (req, res) {
+    try {
+        const body = req.body;
+        const [user] = await User.findUser(body.businessEmail);
         if (user && user.userId) {
             if (bcrypt.compareSync(body.password, user.password)) {
-                    delete user.password;
-                    const token = jwt.sign(user, process.env.JWT_SECRET, {
-                        expiresIn: process.env.EXPIRE_IN
-                    });
-                    res.status(200).json({
-                        success: true,
-                        message: 'Logged in successfully',
-                        userInfo: user,
-                        authToken: token
-                    });
-            } else
-                res.status(401).json({
-                    success: false,
-                    message: 'Login Failed!'
+                delete user.password;
+                const token = jwt.sign(user, process.env.JWT_KEY, {
+                    expiresIn: process.env.EXPIRE_IN
                 });
-        } else
-            res.status(401).json({
-                success: false,
-                message: 'Login Failed!'
-            });
-        if (err)
-            res.status(500).json({
-                success: false,
-                message: 'Something went wrong,Error while authenticating user!'
-            });
-    });
+                return res.status(200).json({ success: true, message: 'Logged in successfully', userInfo: user, authToken: token });
+            } else {
+                return res.status(401).json({ success: false, message: 'Login failed.' });
```

Comment:
It could be that this is fixed in one of the other PRs, but to be sure: It would be preferable to not make any difference in the error code and message between a user that:
1. Enters a valid user name and invalid password.
2. Enters an invalid user name.

If there's a difference, a malicious user would know whether a user name is valid, which simplifies brute-force attacks.

Reply:
yes this 401 error code will reveal malicious user to know the scenario. so changing the error code to 409.
please suggest if this error code does not match.

Reply:
`409` indicates `Conflict` (e.g. something like Already Exists). My point here was pointed at the fact that there's a difference in the error __message__ based on whether the user exists or not. I would recommend returning `401` with a message like `Login failed.` both when the password is wrong and when a non-existing user name has been entered.